src/ReactWrapper.jsx

```diff
@@ -208,7 +208,10 @@ class ReactWrapper {
       throw new Error('ReactWrapper::unmount() can only be called on the root');
     }
     this.single('unmount', () => {
-      this.component.setState({ mount: false });
+      this.component.setState({
+        mount: false,
+        lastRender: this.component._reactInternalInstance._renderedComponent.getPublicInstance(),
```

Comment:
Sorry I wasn't being clear. The `WeakMap` would be a module-level variable accessible to all instances via closure - ie, `const instanceMap = new WeakMap(); class ShallowWrapper { … }`